include/xrpl/protocol/detail/ledger_entries.macro

```diff
@@ -436,3 +439,26 @@ LEDGER_ENTRY(ltCREDENTIAL, 0x0081, Credential, ({
     {sfPreviousTxnID,        soeREQUIRED},
     {sfPreviousTxnLgrSeq,    soeREQUIRED},
 }))
+
+/** A ledger object representing a single asset vault.
+
+    \sa keylet::mptoken
+ */
+LEDGER_ENTRY(ltVAULT, 0x0082, Vault, ({
+    {sfPreviousTxnID,        soeREQUIRED},
+    {sfPreviousTxnLgrSeq,    soeREQUIRED},
+    {sfSequence,             soeREQUIRED},
+    {sfOwnerNode,            soeREQUIRED},
+    {sfOwner,                soeREQUIRED},
+    {sfAccount,              soeREQUIRED},
+    {sfData,                 soeDEFAULT},
+    {sfAsset,                soeREQUIRED},
+    {sfAssetTotal,           soeDEFAULT},
+    {sfAssetAvailable,       soeDEFAULT},
+    {sfAssetMaximum,         soeDEFAULT},
+    {sfLossUnrealized,       soeDEFAULT},
+    {sfMPTokenIssuanceID,    soeREQUIRED}, // sfShare
```

Comment:
- `ShareTotal` is not going to be implemented. We decided to just use the bookkeeping in `mptIssuance(vault.sfMPTIssuanceID).sfOutstandingAmount`. However, the RPC for `ledger_object` (or whatever it's called) needs to join that field in the result.
- Right now, there is only one `WithdrawalPolicy`. IMO, it can be added (and tested) once a second is introduced.
- `PermissionedDomainID` is planned for v1, but isn't yet available in `develop`. Its PR (#5161) needs to be merged with this branch.